pkg/component/logging/vali/vali.go

```diff
@@ -0,0 +1,1104 @@
+			Namespace: v.namespace,
+			Annotations: map[string]string{
+				"nginx.ingress.kubernetes.io/configuration-snippet": "proxy_set_header X-Scope-OrgID operator;",
+			},
+			Labels: getLabels(),
+		},
+		Spec: networkingv1.IngressSpec{
+			TLS: []networkingv1.IngressTLS{
+				{
+					SecretName: secretName,
+					Hosts:      []string{v.values.ValiHost},
+				},
+			},
+			Rules: []networkingv1.IngressRule{
+				{
+					Host: v.values.ValiHost,
+					IngressRuleValue: networkingv1.IngressRuleValue{
+						HTTP: &networkingv1.HTTPIngressRuleValue{
+							Paths: []networkingv1.HTTPIngressPath{
+								{
+									Backend: networkingv1.IngressBackend{
+										Service: &networkingv1.IngressServiceBackend{
+											Name: serviceName,
+											Port: networkingv1.ServiceBackendPort{
+												Number: serviceRBACProxyPortNumber,
+											},
+										},
+									},
+									Path:     backendPath,
+									PathType: &pathType,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if version.ConstraintK8sGreaterEqual122.Check(v.k8sVersion) {
+		ingress.Spec.IngressClassName = pointer.String(v.values.IngressClass)
+	} else {
+		ingress.Annotations["kubernetes.io/ingress.class"] = v.values.IngressClass
+	}
+
+	return ingress
+}
+
+func (v *vali) getService() *corev1.Service {
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "logging",
+			Namespace:   v.namespace,
+			Labels:      getLabels(),
+			Annotations: map[string]string{},
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{
+					Port:       serviceValiPortNumber,
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.FromInt(serviceValiPortNumber),
+					Name:       valiMetricsPortName,
+				},
+			},
+			Selector: getLabels(),
+		},
+	}
+
+	if v.isRBACProxyEnabled() {
+		svc.Spec.Ports = append(svc.Spec.Ports, []corev1.ServicePort{
+			{
+				Port:       serviceRBACProxyPortNumber,
+				Protocol:   corev1.ProtocolTCP,
+				TargetPort: intstr.FromInt(serviceRBACProxyPortNumber),
+				Name:       "external",
+			},
+			{
+				Port:       serviceTelegrafPortNumber,
+				Protocol:   corev1.ProtocolTCP,
+				TargetPort: intstr.FromInt(serviceTelegrafPortNumber),
+				Name:       telegrafName,
+			},
+		}...)
+	}
+
+	switch v.values.ClusterType {
+	case "seed":
+		svc.Annotations["networking.resources.gardener.cloud/from-all-seed-scrape-targets-allowed-ports"] = fromAllSeedScrapeTargetsAllowedPorts(v.isRBACProxyEnabled())
+	case "shoot":
+		svc.Annotations["networking.resources.gardener.cloud/from-all-scrape-targets-allowed-ports"] = fromAllScrapeTargetsAllowedPorts(v.isRBACProxyEnabled())
+		svc.Annotations[resourcesv1alpha1.NetworkingPodLabelSelectorNamespaceAlias] = v1beta1constants.LabelNetworkPolicyShootNamespaceAlias
+		svc.Annotations[resourcesv1alpha1.NetworkingNamespaceSelectors] = `[{"matchLabels":{"kubernetes.io/metadata.name":"garden"}}]`
+	}
```

Comment:
Can you use something like [this](https://github.com/gardener/gardener/blob/a89de8de8dabc8a28757c0313aee882c7de415f9/pkg/component/kubestatemetrics/resources.go#L169-L173) instead? Generally, use the helper functions from `pkg/utils/gardener`.